### DISCUSSION TITLE
Save session after it was created

### DISCUSSION THREAD

**Discussion by mogambro** [zellij-org/zellij#3631] [General]
Hey there!

First off, let me say: I love zellij!

Secondly: When I reopen a session it always opens in the way I created it many weeks ago.

In between I opened other tabs, panes etc. I renamed tabs... but it never saves any of that. It always opens in the correct folder, but just my two panes.

So: How can I manually save/overwrite a session?

Best regards,
Nerdachse aka Tom

**Comment by imsnif** [comment]
Could you describe more how you open sessions? These should be updated automatically, and if it doesn't work for you it might be an issue. Would be helpful to know exactly what you are doing step-by-step (even if it seems obvious to you). Thanks!

**Reply by mogambro** [reply]
Of course, sorry for not doing that from the get go.

I resurrect a session (it has two panes), then work in and with it. When I am done, I kill my terminal (using swaywm and wezterm).

I get asked "Really kill this window and all contained tabs and panes". I confirm.

Zellij keeps running in the background, I see that because in some tabs I ran some commands like "cargo run --release" and the app still runs.

When I want to continue working, I can attach to the still open session. At some point I turn my PC off and the next time I resurrect my session.

I can of course give screenshots if wanted.

**Reply by mogambro** [reply]
@imsnif Hey! I am still facing this problem. It may be totally a layer 8 problem, pebkac if you want, but I am eager to learn. Any ideas what my problem is? (Apart from creating dumb discussions on github that is ;))

**Reply by Lockszmith-GH** [reply]
All of this tied to session resurrection feature.

Documentation: https://zellij.dev/documentation/session-resurrection

Relevant past discussions:
#2919 was about how to stop it.
#2282 is about how to save the current layout.

**Comment by HungryJoe** [comment]
Hi, I came across this thread with the exact same question as OP: Why do my resurrected sessions have outdated layouts?

While I appreciate @Lockszmith-GH's suggestion, to me this is a work-around; ideally, my sessions would automatically be resurrected with the layout they had when I closed them most recently. Instead, I only get this behavior when, per #2282, I run 
```bash
zellij --session <ID> action dump-layout path/to/zellij/cache/0.43.1/session_info/<ID>/session-layout.kdl
```
just before closing the session `<ID>`.

Per @imsnif's question above, I open and close sessions in the following way:
1. `zellij -l welcome`
2. Press `<tab>, <tab>` to get to the "tab" that says *Resurrect Session*
3. Select a session's ID with arrow keys
4. Press `<Enter>` to resurrect the session
5. After I'm done using the session, I press `<C-q>` (`Ctrl+q` on a Mac) to close the session and quit Zellij.

#### Computing Environment
* OS: **macOS 15.6.1**
* `uname -mrs`: **Darwin 24.6.0 x86_64**
* `zellij --version`: **zellij 0.43.1**
* Terminal: **iTerm2 3.6.2**
* Zellij installed and configured via **Nix home-manager**
* Shell: **fish 4.0.8**

##### Home Manager Config
```nix collapsible
  zellij = {
    enable = true;
    enableFishIntegration = true;
    settings = {
      theme = "catppuccin-latte";
      show_startup_tips = false;
      pane_viewport_serialization = true;
    };
  };
```
